### ISSUE TITLE
NCPA don't start after reboot

### ISSUE THREAD

**Issue by metabsd** [NagiosEnterprises/ncpa#365]
Hi,

I have that problem on multiple server. RHEL 6 - RHEL 7.

The service is active on runlevel 3 like the other.  

```
chkconfig --list | grep -i ncpa
ncpa_listener  	0:off	1:off	2:on	3:on	4:on	5:on	6:off
ncpa_passive   	0:off	1:off	2:on	3:on	4:on	5:on	6:off
```
That server just restart  

```
11:14:44 up  1:12,  2 users,  load average: 1.03, 1.09, 1.06
```
The status of the service return that   

```
/etc/init.d/ncpa_listener status
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/cx_Freeze/initscripts/Console.py", line 27, in <module>
  File "ncpa_listener.py", line 83, in <module>
  File "/root/ncpa/agent/ncpadaemon.py", line 80, in main
  File "/root/ncpa/agent/ncpadaemon.py", line 231, in status
ValueError: invalid literal for int() with base 10: ''
NCPA Listener 
```

Nothing run  
```
ps auxwww | grep -i ncpa | grep -v grep
```

```
/etc/init.d/ncpa_listener start
Starting NCPA Listener: pidfile /usr/local/ncpa/var/run/ncpa_listener.pid contains a non-integer value
                                                           [FAILED]
```

```cat /usr/local/ncpa/var/run/ncpa_listener.pid
ll /usr/local/ncpa/var/run/ncpa_listener.pid
-rw------- 1 nagios nagios 0 Aug 11 10:01 /usr/local/ncpa/var/run/ncpa_listener.pid
```

I remove the pid file.  
`rm -f /usr/local/ncpa/var/run/ncpa_listener.pid`

Now I'm able to start ncpa_listener.  
```
/etc/init.d/ncpa_listener start
Starting NCPA Listener:                                    [  OK  ]
```

```
/etc/init.d/ncpa_listener status
service is running (pid 8700)
NCPA Listener 
```

**Comment by jomann09** [comment]
Interesting I will look into this, will try to get it fixed in the next version - looks like it is not able to read an empty or partially corrupted pid file (and it should be able to figure out what is going on if that is the case)

**Comment by metabsd** [comment]
Another server just reboot and NCPA don't restart properly.

I start seeing unknown alert about that server. I ssh in and see ncpa_listener not working.
```
[root@slpmdm0668 ~]# systemctl status ncpa_listener
● ncpa_listener.service - LSB: This manages the NCPA Listener service
   Loaded: loaded (/etc/rc.d/init.d/ncpa_listener; bad; vendor preset: disabled)
   Active: active (exited) since Thu 2017-08-17 11:15:00 EDT; 9h ago
     Docs: man:systemd-sysv-generator(8)
  Process: 1175 ExecStart=/etc/rc.d/init.d/ncpa_listener start (code=exited, status=0/SUCCESS)

Aug 17 11:14:58 slpmdm0668 systemd[1]: Starting LSB: This manages the NCPA Listener service...
Aug 17 11:15:00 slpmdm0668 ncpa_listener[1175]: Starting NCPA Listener: [  OK  ]
Aug 17 11:15:00 slpmdm0668 systemd[1]: Started LSB: This manages the NCPA Listener service.
```

I tail ncpa_listener.log and see that at the end.

```
2017-08-17 11:14:21,084 31051 INFO stopped
2017-08-17 11:15:00,042 2852 INFO started
2017-08-17 11:15:05,071 2852 ERROR stopping with an exception
Traceback (most recent call last):
  File "/root/ncpa/agent/ncpadaemon.py", line 197, in start
  File "ncpa_listener.py", line 43, in run
  File "/root/ncpa/agent/listener/database.py", line 57, in run_db_maintenance
OperationalError: database is locked
2017-08-17 11:15:05,072 2852 INFO stopped
```
I run `systemctl restart ncpa_listener` and he start normally.

Systemd don't see the unit as normaly. I also `chkconfig --list | grep ncpa_listener` and see that output.

```
[root@slpmdm0668 log]# chkconfig --list | grep ncpa_listener

Note: This output shows SysV services only and does not include native
      systemd services. SysV configuration data might be overridden by native
      systemd configuration.

      If you want to list systemd services use 'systemctl list-unit-files'.
      To see services enabled on particular target use
      'systemctl list-dependencies [target]'.

ncpa_listener   0:off   1:off   2:on    3:on    4:on    5:on    6:off
```

ncpa_listener don't exist as unit-files in systemd.

This is the output of `systemctl list-dependencies ncpa_listener`
```
ncpa_listener.service
● ├─system.slice
● └─basic.target
●   ├─microcode.service
●   ├─rhel-autorelabel-mark.service
●   ├─rhel-autorelabel.service
●   ├─rhel-configure.service
●   ├─rhel-dmesg.service
●   ├─rhel-loadmodules.service
●   ├─selinux-policy-migrate-local-changes@targeted.service
●   ├─paths.target
●   ├─slices.target
●   │ ├─-.slice
●   │ └─system.slice
●   ├─sockets.target
●   │ ├─dbus.socket
●   │ ├─dm-event.socket
●   │ ├─rpcbind.socket
●   │ ├─systemd-initctl.socket
●   │ ├─systemd-journald.socket
●   │ ├─systemd-shutdownd.socket
●   │ ├─systemd-udevd-control.socket
●   │ └─systemd-udevd-kernel.socket
●   ├─sysinit.target
●   │ ├─dev-hugepages.mount
●   │ ├─dev-mqueue.mount
●   │ ├─dmraid-activation.service
●   │ ├─kmod-static-nodes.service
●   │ ├─lvm2-lvmetad.socket
●   │ ├─lvm2-lvmpolld.socket
●   │ ├─lvm2-monitor.service
●   │ ├─plymouth-read-write.service
●   │ ├─plymouth-start.service
●   │ ├─proc-sys-fs-binfmt_misc.automount
●   │ ├─sys-fs-fuse-connections.mount
●   │ ├─sys-kernel-config.mount
●   │ ├─sys-kernel-debug.mount
●   │ ├─systemd-ask-password-console.path
●   │ ├─systemd-binfmt.service
●   │ ├─systemd-firstboot.service
●   │ ├─systemd-hwdb-update.service
●   │ ├─systemd-journal-catalog-update.service
●   │ ├─systemd-journal-flush.service
●   │ ├─systemd-journald.service
●   │ ├─systemd-machine-id-commit.service
●   │ ├─systemd-modules-load.service
●   │ ├─systemd-random-seed.service
●   │ ├─systemd-sysctl.service
●   │ ├─systemd-tmpfiles-setup-dev.service
●   │ ├─systemd-tmpfiles-setup.service
●   │ ├─systemd-udev-trigger.service
●   │ ├─systemd-udevd.service
●   │ ├─systemd-update-done.service
●   │ ├─systemd-update-utmp.service
●   │ ├─systemd-vconsole-setup.service
●   │ ├─cryptsetup.target
●   │ ├─local-fs.target
●   │ │ ├─-.mount
●   │ │ ├─boot.mount
●   │ │ ├─home.mount
●   │ │ ├─opt.mount
●   │ │ ├─rhel-import-state.service
●   │ │ ├─rhel-readonly.service
●   │ │ ├─systemd-remount-fs.service
●   │ │ ├─tmp.mount
●   │ │ ├─var-log-audit.mount
●   │ │ ├─var-log.mount
●   │ │ ├─var-tmp.mount
●   │ │ └─var.mount
●   │ └─swap.target
●   │   └─dev-mapper-rootvg\x2dlv00.swap
●   └─timers.target
●     └─systemd-tmpfiles-clean.timer
```

```
[root@slpmdm0668 log]# systemctl status ncpa_listener -l
● ncpa_listener.service - LSB: This manages the NCPA Listener service
   Loaded: loaded (/etc/rc.d/init.d/ncpa_listener; bad; vendor preset: disabled)
   Active: active (running) since Thu 2017-08-17 21:11:29 EDT; 6min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 19867 ExecStop=/etc/rc.d/init.d/ncpa_listener stop (code=exited, status=0/SUCCESS)
  Process: 19877 ExecStart=/etc/rc.d/init.d/ncpa_listener start (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/ncpa_listener.service
           └─19886 /usr/local/ncpa/ncpa_listener --start

Aug 17 21:11:28 slpmdm0668 systemd[1]: Starting LSB: This manages the NCPA Listener service...
Aug 17 21:11:29 slpmdm0668 systemd[1]: Started LSB: This manages the NCPA Listener service.
Aug 17 21:11:29 slpmdm0668 ncpa_listener[19877]: Starting NCPA Listener: [  OK  ]
```

At that line `Loaded: loaded (/etc/rc.d/init.d/ncpa_listener; bad; vendor preset: disabled)`

I see after ncpa_listener; bad... Normally I should see enabled and that confirm the service is enabled at the start.

Please fix that shortly because we have to restart manually a lot of ncpa_listener manually.

Thank you!

**Comment by metabsd** [comment]
I have another example of problem after server reboot.

Someone have to reboot a server (Normal reboot)

```
2017-08-17 10:44:04,436 2140 ERROR stopping with an exception
Traceback (most recent call last):
  File "/root/ncpa/agent/ncpadaemon.py", line 197, in start
  File "ncpa_listener.py", line 43, in run
  File "/root/ncpa/agent/listener/database.py", line 57, in run_db_maintenance
OperationalError: database is locked
2017-08-17 10:44:04,437 2140 INFO stopped
2017-08-17 11:08:12,314 1542 INFO started
2017-08-17 11:08:17,969 1542 ERROR stopping with an exception
Traceback (most recent call last):
  File "/root/ncpa/agent/ncpadaemon.py", line 197, in start
  File "ncpa_listener.py", line 43, in run
  File "/root/ncpa/agent/listener/database.py", line 57, in run_db_maintenance
OperationalError: database is locked
2017-08-17 11:08:17,970 1542 INFO stopped
```

The server restart and the service don't restart at boot as expected.

I manually `systemcte restart ncpa_listener` and the service start correctly.

I have the same word bad after ncpa_listener
```
[root@slqwas0613 log]# systemctl status ncpa_listener
● ncpa_listener.service - LSB: This manages the NCPA Listener service
   Loaded: loaded (/etc/rc.d/init.d/ncpa_listener; bad; vendor preset: disabled)
   Active: active (running) since Thu 2017-08-17 21:24:18 EDT; 8s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 1768 ExecStop=/etc/rc.d/init.d/ncpa_listener stop (code=exited, status=0/SUCCESS)
  Process: 1779 ExecStart=/etc/rc.d/init.d/ncpa_listener start (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/ncpa_listener.service
           └─1790 /usr/local/ncpa/ncpa_listener --start

Aug 17 21:24:18 slqwas0613 systemd[1]: Starting LSB: This manages the NCPA Listener service...
Aug 17 21:24:18 slqwas0613 ncpa_listener[1779]: Starting NCPA Listener: [  OK  ]
Aug 17 21:24:18 slqwas0613 systemd[1]: Started LSB: This manages the NCPA Listener service.
```

Wierd!

**Comment by jomann09** [comment]
Thanks for the update, I will try to get a fix out for this as soon as possible.

**Comment by acefreakz** [comment]
@metabsd I too having the same issue, it just wont start after server reboot. Manually restart it via systemd works fine tho.

**Comment by sampath4970** [comment]
Is there any fix released for this yet?

**Comment by jomann09** [comment]
For the database lock issue or the pid file being empty?

**Comment by jomann09** [comment]
Empty PID fix should be here cab54698e0a9338eac5fa37dbb138448a6a8a5d3 and the update for the db having issues should also be working. I am going to build a test version of 2.0.5 and hopefully release it on Monday. I will continue working on 2.1 though, which should be released shortly (couple weeks max)

**Comment by jomann09** [comment]
Closing this since the original 2 issues should be taken care of in 2.0.5.
The remaining change for systemd is being tracked in #349 and will probably be bumped into 2.1 now.
